src/inscription.rs

```diff
@@ -99,19 +116,35 @@ impl<'a> InscriptionParser<'a> {
 
   fn parse_inscription(&mut self) -> Result<Option<Inscription>> {
     if self.advance()? == Instruction::Op(opcodes::all::OP_IF) {
-      let content = self.advance()?;
-
-      let content = if let Instruction::PushBytes(bytes) = content {
+      let media_type = if let Instruction::PushBytes(bytes) = self.advance()? {
         str::from_utf8(bytes).map_err(InscriptionError::Utf8Decode)?
       } else {
         return Err(InscriptionError::InvalidInscription);
       };
 
+      let content = if let Instruction::PushBytes(bytes) = self.advance()? {
+        bytes
+      } else {
+        return Err(InscriptionError::InvalidInscription);
+      };
+
+      let inscription = match media_type {
+        "text/plain;charset=utf-8" => Some(Inscription::Text(
+          str::from_utf8(content)
+            .map_err(InscriptionError::Utf8Decode)?
+            .into(),
+        )),
+        "image/png" => Some(Inscription::Png(content.to_vec())),
+        _ => {
+          return Err(InscriptionError::InvalidInscription);
+        }
```

Comment:
I think we should have an unknown inscription type:

```suggestion
        _ => Inscription::Unknown { media_type, content },
```

Reply:
This is so, if we later need to have a canonical view of all inscriptions, which inscriptions exist doesn't depend on what mime type they have.

Reply:
Also add a test that unknown inscriptions are parsed from scripts correctly.